solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2424,6 +2430,34 @@ public PlacementPluginFactory<? extends PlacementPluginConfig> getPlacementPlugi
   public void runAsync(Runnable r) {
     coreContainerAsyncTaskExecutor.submit(r);
   }
+
+  public static void setWeakStringInterner() {
+    boolean enable = "true".equals(System.getProperty("use.str.intern", "true"));
```

Comment:
This should be prefixed with solr.

Maybe instead of true/false it would make sense to do strong/weak/off? No real opinion here, talking through ideas out loud to see how they sound... probably too much complexity and not worth it, but I'll leave that up to you.